extensions/2.0/Khronos/KHR_lights_environment/README.md

```diff
@@ -0,0 +1,174 @@
+
+# KHR_lights_environment
+
+## Contributors
+
+* Norbert Nopper, <mailto:nopper@ux3d.io>
+* Rickard Sahlin, <mailto:rickard.sahlin@ikea.com>
+* Gary Hsu, Microsoft, <mailto:garyhsu@microsoft.com>
+* Mike Bond, Adobe, <mailto:mbond@adobe.com>
+* Ben Houston, ThreeKit, <mailto:bhouston@threekit.com>
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+This extension may use KHR_texture_basisu, KHR_texture_float_bc6h or KHR_texture_float_f16  
```

Comment:
I would suggest we only pick a single texture format for HDR data that is going to be standard, such as float_bc6h (which is specifically designed for the HDR use case) and only support that one.  It will make implementation easier and also conformance testing.  People can just transcode to their preferred format (of which there are many.). There is an 11KB JS decoder for BC6H which we can recommend.  For example Filament loves R11G11B10F: https://twitter.com/romainguy/status/1367926957267021824

Reply:
> There is an 11KB JS decoder for BC6H which we can recommend.

Did you mean [7kB WASM decoders](https://github.com/KhronosGroup/BC6H-Decoder-WASM/tree/main/build), or is there another implementation?

Reply:
Sorry, that was the one I was referring to -- you had referred me to it earlier: https://github.com/KhronosGroup/BC6H-Decoder-WASM/tree/main/build

Reply:
Saying now that a format is going to be standard is like taking up that magic 8-ball and asking it a question :-)

Jokes aside, I do really believe that we shall create a pragmatic yet flexible solution.
We should look outside the WebGL (and browser) usage and aim for the future.
The wasm libraries, while useful as a reference, will only do so much if your viewer is not JS based - it's not a silver bullet.
For our viewer in Java it can only serve as a reference - I'd rather start with uncompressed formats and add compressed formats using extensions such as the KHR_texture_basisu

Reply:
Assuming that the Java viewer runs on a desktop, BC6H support is already there, in hardware. Besides, the compiled WASM modules could be auto-translated to C or compiled directly with LLVM. JavaScript is irrelevant here.

Uncompressed and non-universal formats are anything but pragmatic (although still fine as a vendor-specific use case).

Reply:
Since BC6H is a texture format that needs support on runtime platform implementations need to have a fallback.
Implementing this fallback will be an effort - if your target is a browser and your language is javascript, calling the wasm module containing the transcoder will be trivial. 
Hence, I see js as relevant for the implementation effort.

For other languages the threshold will vary depending on your architecture and design goals.

>Uncompressed and non-universal formats are anything but pragmatic (although still fine as a vendor-specific use case).

What do you mean by non-universal formats?

Reply:
As a standard, WebAssembly goes far beyond the web (despite its name). Standalone binary modules could be further compiled to native code or embedded as-is in almost any runtime environment via [Wasmer](https://wasmer.io/).

Calling exported Wasm functions from Java, for example, is almost as trivial as doing that from JS:
https://github.com/wasmerio/wasmer-java#example

Formats like RGB16F or RGB32F are:
- uncompressed by design and remain uncompressed on the GPU (sub-optimal bandwidth usage);
- don't compress well with zlib or zstd without domain-specific prepass (sub-optimal transmission);
- require per-pixel work (converting to RGBA16F) on a fair amount of platforms (sub-optimal loading, non-universal).

As such, I'd rather prefer standardizing on a dedicated floating-point lossless compression scheme that may decode to RGB16F/RGB32F if we can confirm that some environment maps absolutely need that level of precision.

Reply:
> As a standard, WebAssembly goes far beyond the web (despite its name). Standalone binary modules could be further compiled to native code or embedded as-is in almost any runtime environment via [Wasmer](https://wasmer.io/).

I think that's taking it a bit far - tbh Wasm was created with web and javascript focus in mind. 
Sure it's a solution that _can_ be used in a multitude of usecases and enviroments - but that does not mean that we as a group shall see a wasm module as 'effort free' library to implement.  

Using wasmer to call make a JNI call to wasm _seams easy enough - however,  it may not suit the architecture of the project. 
You would be introducing another 3rd party library and possibly breaking design goals by introducing JNI calls (which wont be debuggable in the JRE)

All I'm saying is that it's not as simple as:
- "Look, implementations will get this functionallity for free since it's available in a wasm lib."

Reply:
This extension now requires KHR_texture_ktx - I see the question of what formats and how as resolved in this extension.
Please, let's continue discussion of relevant formats in KHR_texture_ktx.

Reply:
> I think that's taking it a bit far - tbh Wasm was created with web and javascript focus in mind.

The scope has grown since then.

> Look, implementations will get this functionality for free since it's available in a wasm lib.

Porting its source (~1k LoC of simple math) to plain C or Java is so easy that I could do that over a weekend (not the current one, obviously). Additionally, I've got a GLSL implementation of it ready (not public yet) that performs the decoding in a single fragment shader. So, yes, implementations will get this for free eventually. Wasmer is just another approach as it could use the modules as-is.

Reply:
I guess we just have to agree to disagree :-)